src/ExpressionEvaluator/Core/Source/ExpressionCompiler/EvaluationContextBase.cs

```diff
@@ -110,6 +110,8 @@ internal string GetErrorMessageAndMissingAssemblyIdentities(DiagnosticBag diagno
 
         internal abstract ImmutableArray<AssemblyIdentity> GetMissingAssemblyIdentities(Diagnostic diagnostic);
 
+        internal abstract bool HasReferenceToAny(ImmutableArray<AssemblyIdentity> identities);
```

Comment:
It accesses the EvaluationContext.Compilation (which is language specific).  We could pull that up to the base class and make it a CommonCompilation, but it seemed like that would introduce more casting elsewhere (although I didn't try it).